server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -4048,7 +4048,10 @@ public UserVmVO doInTransaction(TransactionStatus status) throws InsufficientCap
                         vm.setDetail(VmDetailConstants.FIRMWARE, "efi");
                         s_logger.info("guestOS is OSX : overwrite root disk controller to scsi, use smc and efi");
                     } else {
-                        String controllerSetting = _configDao.getValue("vmware.root.disk.controller");
+                        String controllerSetting = _configDao.getValue(Config.VmwareRootDiskControllerType.key());
+                        if (StringUtils.isEmpty(controllerSetting)) {
+                            controllerSetting = Config.VmwareRootDiskControllerType.getDefaultValue();
+                        }
```

Comment:
converting this setting to modern ConfigKey<> would be more failsafe and concise.

Reply:
ok, i see. that is another type of issue than; a plugin specific setting used in generic code. Don't see this as a minus 1 by the way.

Reply:
@shwstppr can we use this instead? 
```suggestion
String controllerSetting = StringUtils.defaultIfEmpty(_configDao.getValue(Config.VmwareRootDiskControllerType.key()),
                                Config.VmwareRootDiskControllerType.getDefaultValue());
```

Reply:
> @DaanHoogland wanted to do that but config is used in UserVmManagerImpl and `plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/manager/VmwareManagerImpl.java` so not sure where to place the ConfigKey to access at both places

@shwstppr Agree with @DaanHoogland , better move it to ConfigKey. There are some classes where you can add this config (Eg. StorageManager, can also check Volume / Hypervisor).